File: pset6/mario/less/mario-less.py

# Define Recursion for half-pyramid
def block(layer):

    if layer == 0:
        return

    block(layer - 1)

    print(" " * (spaceNum - layer), end="")
    print("#" * layer, end="")
    print()


# Prompting the user to type in positive interger
while True:

    try:
        h = int(input("Height: "))

    # Re-prompting the user to type in valid interger
    except:
        continue

    if 1 <= h <= 8:
        break

    '''Another method:
    from cs50 import get_int
    h = get_int("Height: ")'''

# Build half-pyramid
spaceNum = h
block(h)

File: pset1/mario/less/mario-less.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{
    //Prompt user for positive integer
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
 
 
    //Print out right-side pyramid
    for (int i = 0; i < n; i++)
    {
        //Print out space
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }

        //Print out # brick
        for (int w = 0; w < i + 1; w++)
        {
            printf("#");
        }
        printf("\n");
    }
}